include/aws/common/cpu_usage_sampler.h

```diff
@@ -0,0 +1,52 @@
+#ifndef AWS_COMMON_CPU_USAGE_SAMPLER_H
+#define AWS_COMMON_CPU_USAGE_SAMPLER_H
+
+/**
+ * Copyright Amazon.com, Inc. or its affiliates. All Rights Reserved.
+ * SPDX-License-Identifier: Apache-2.0.
+ */
+
+#include <aws/common/common.h>
+#include <aws/common/math.h>
+
+/**
+ * A struct that contains the CPU sampler for this platform.
+ * Currently only Linux is supported.
+ *
+ * Note: Must be freed from memory using aws_cpu_sampler_destroy when finished.
+ */
+struct aws_cpu_sampler;
+
+AWS_EXTERN_C_BEGIN
+
+/**
+ * Creates a new CPU sampler using the provided allocator, or will return NULL if there is an error.
+ *
+ * Note: On unsupported platforms, the CPU sampler returned will return AWS_OP_ERR when calling
+ * aws_cpu_sampler_get_sample. You will still need to call aws_cpu_sampler_clean_up when finished
+ * to free the memory even for unsupported platforms.
+ */
+AWS_COMMON_API
+struct aws_cpu_sampler *aws_cpu_sampler_new(struct aws_allocator *allocator);
```

Comment:
I would name this something like aws_system_cpu_sampler to be clear it's querying the system, and not this process or this library